packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,658 @@
+    class Import extends Resource implements IHealthCheck {
+      public readonly healthCheckId = healthCheckId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly healthCheckId: string;
+
+  /**
+   * Creates a new health check.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param props the properties of the new health check.
+   * @returns a reference to the newly created health check.
+   */
+  constructor(scope: Construct, id: string, props: HealthCheckProps) {
+    super(scope, id);
+
+    validateProperties(props);
+
+    const resource = new CfnHealthCheck(this, 'Resource', {
+      healthCheckConfig: {
+        type: props.type,
+        alarmIdentifier: props.alarmIdentifier,
+        childHealthChecks: props.childHealthChecks?.map((childHealthCheck) => childHealthCheck.healthCheckId),
+        enableSni: props.enableSNI ?? getDefaultEnableSNIForType(props.type),
+        failureThreshold: props.failureThreshold ?? getDefaultFailureThresholdForType(props.type),
+        fullyQualifiedDomainName: props.fqdn,
+        healthThreshold: props.healthThreshold ?? getDefaultHealthThresholdForType(props),
+        insufficientDataHealthStatus: props.insufficientDataHealthStatus ?? getDefaultInsufficientDataHealthStatusForType(props.type),
+        inverted: props.inverted ?? false,
+        ipAddress: props.ipAddress,
+        measureLatency: props.measureLatency ?? getDefaultMeasureLatencyForType(props.type),
+        port: props.port ?? getDefaultPortForType(props.type),
+        regions: props.regions,
+        requestInterval: props.requestInterval?.toSeconds() ?? getDefaultRequestIntervalForType(props.type)?.toSeconds(),
+        resourcePath: props.resourcePath ?? getDefaultResourcePathForType(props.type),
+        routingControlArn: props.routingControl,
+        searchString: props.searchString,
+      },
+    });
+
+    this.healthCheckId = resource.ref;
+  }
+}
+
+const validationRules: Record<HealthCheckType, ((props: HealthCheckProps) => void)[]> = {
+  [HealthCheckType.HTTP]: [
+    ruleAlarmIdentifierIsNotAllowed,
+    ruleEnableSNIIsNotAllowed,
+    ruleSearchStringIsNotAllowed,
+    ruleChildHealthChecksIsNotAllowed,
+    ruleRoutingControlIsNotAllowed,
+    ruleHealthThresholdIsNotAllowed,
+    validateFqdn,
+    validateIpAddress,
+    validateEitherFqdnOrIpAddressRequired,
+    validateRequestInterval,
+    validateFailureThreshold,
+    validateRegions,
+  ],
+  [HealthCheckType.HTTPS]: [
+    ruleAlarmIdentifierIsNotAllowed,
+    ruleSearchStringIsNotAllowed,
+    ruleChildHealthChecksIsNotAllowed,
+    ruleRoutingControlIsNotAllowed,
+    ruleHealthThresholdIsNotAllowed,
+    validateFqdn,
+    validateIpAddress,
+    validateEitherFqdnOrIpAddressRequired,
+    validateRequestInterval,
+    validateFailureThreshold,
+    validateRegions,
+  ],
+  [HealthCheckType.HTTP_STR_MATCH]: [
```

Comment:
I appreciate the fact that you added all these validations on CDK side, however, this also makes me a bit worried at the same time. If CloudFormation decides to change one or some of the validation rules, we need to make changes accordingly and given there are so many, it's not straightforward to update this.

How many of these validations rule are essential? Can we only validate on the essential ones and leave the rest to CFN?